multinsclient/client.go

```diff
@@ -0,0 +1,177 @@
+package multinsclient
+
+import (
+	"fmt"
+	"hash/fnv"
+
+	"go.temporal.io/sdk/client"
+)
+
+// Namespace represents a client and a namespace for connecting.
+type Namespace struct {
+	// Options for the client. The Namespace field must be populated. The options
+	// are never used to connect to a client if Disabled is true.
+	ClientOptions client.Options
+	// If true, this namespace will never have a client created for it.
+	Disabled bool
+}
+
+// Client contains clients for different namespaces.
+type Client struct {
+	allNamespaces           []string
+	enabledNamespaces       []string
+	enabledNamespaceClients map[string]client.Client
+	hasher                  func(workflowID string, max int) int
+}
+
+// Options for New.
+type Options struct {
+	// Namespaces this client will work with. The order and count and names should
+	// remain the same always so the hashing is somewhat consistent. At least one
+	// enabled namespace is required.
+	//
+	// See Client.NamespaceFor for details on namespace choice.
+	Namespaces []Namespace
+
+	// Hasher that chooses a number from 0 (inclusive) to max (exclusive) for the
+	// given workflow ID. If unset, DefaultHasher is used.
+	Hasher func(workflowID string, max int) int
+
+	// If true, enabled namespaces will create clients that won't connect until
+	// first used. Otherwise, enabled namespaces will have their clients eagerly
+	// attempt to connect to the server when the client is created.
+	ConnectLazy bool
```

Comment:
Do we really need this? I would take it out for simplicity.

Reply:
I don't know whether they use lazy connectivity or eager connectivity today.